Automattic/woocommerce-payments
# Fix/2593 disabling woopay express avoids disabling direct checkout

Fixes 2593-gh-Automattic/woopay.

#### Changes proposed in this Pull Request









This PR ensures that the WooPay Direct Checkout flow continues to work as intended whether the WooPay Express Checkout button is enabled or disabled on the cart page.

#### Testing instructions





**Regression Test: Ensure WooPay Direct Checkout flow works as intended when the Express Checkout button is enabled on the cart page**
* Ensure the WooPay Direct Checkout flow is enabled on the merchant's site.
* As a merchant, navigate to `Payments > Settings`.
* Scroll down to the `Express checkouts` section and click on the `Customize` button for WooPay.
* Under the `Enable WooPay button on selected pages` section, make sure the `Cart Page` is enabled.
* As a shopper, ensure you are authenticated on WooPay.
* Navigate to the Merchant shop, add an item to your cart, and navigate to the Cart page.
* Click the `Proceed to checkout` button.
* Ensure you are navigated to WooPay checkout.
* Click the `Place order` button and ensure you are able to checkout successfully.
* Repeat these steps once using the wcblocks-cart and once using the shortcode-cart.

**Test: Ensure WooPay Direct Checkout flow works as intended when the Express Checkout button is disabled on the cart page**
* Ensure the WooPay Direct Checkout flow is enabled on the merchant's site.
* As a merchant, navigate to `Payments > Settings`.
* Scroll down to the `Express checkouts` section and click on the `Customize` button for WooPay.
* Under the `Enable WooPay button on selected pages` section, make sure the `Cart Page` is disabled.
* As a shopper, ensure you are authenticated on WooPay.
* Navigate to the Merchant shop, add an item to your cart, and navigate to the Cart page.
* Click the `Proceed to checkout` button.
* Ensure you are navigated to WooPay checkout.
* Click the `Place order` button and ensure you are able to checkout successfully.
* Repeat these steps once using the wcblocks-cart and once using the shortcode-cart.

-------------------


**Post merge**
